bot/converters.py

```diff
@@ -495,22 +495,51 @@ async def convert(self, ctx: Context, argument: str) -> int:
         return duration
 
 
-class UserMentionOrID(UserConverter):
+def _is_an_unambiguous_user_argument(argument: str) -> bool:
+    """Check if the provided argument is a user mention, user id, or username."""
+    has_id_or_mention = bool(IDConverter()._get_id_match(argument) or RE_USER_MENTION.match(argument))
+
+    # Check to see if the author passed a username (a discriminator exists)
+    argument = argument.removeprefix('@')
+    has_username = len(argument) > 5 and argument[-5] == '#'
+
+    return has_id_or_mention or has_username
+
+
+AMBIGUOUS_ARGUMENT_MSG = ("`{argument}` is not a User mention, a User ID or a Username in the format"
+                          " `name#discriminator`.")
+
+
+class UnambiguousUser(UserConverter):
     """
-    Converts to a `discord.User`, but only if a mention or userID is provided.
+    Converts to a `discord.User`, but only if a mention, userID or a username is provided.
 
-    Unlike the default `UserConverter`, it doesn't allow conversion from a name or name#descrim.
-    This is useful in cases where that lookup strategy would lead to ambiguity.
+    Unlike the default `UserConverter`, it doesn't allow conversion from a name.
```

Comment:
Can we switch the explanation from name vs. username to something else, for many it has the same meaning. Maybe make a note that the discriminator has to be included or similar.